internal/pkg/manifest/lb_web_svc_apply_env_test.go

```diff
@@ -0,0 +1,1514 @@
+					String: aws.String("mockEntrypoint"),
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.EntryPoint = &EntryPointOverride{
+					String: aws.String("mockEntrypoint"),
+				}
+			},
+		},
+		"command overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Command = &CommandOverride{
+					String: aws.String("mockCommand"),
+				}
+				svc.Environments["test"].Command = &CommandOverride{
+					String: aws.String("mockCommandTest"),
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.Command = &CommandOverride{
+					String: aws.String("mockCommandTest"),
+				}
+			},
+		},
+		"command not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Command = &CommandOverride{
+					String: aws.String("mockCommand"),
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.Command = &CommandOverride{
+					String: aws.String("mockCommand"),
+				}
+			},
+		},
+		"cpu overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(1024)
+				svc.Environments["test"].CPU = aws.Int(2048)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(2048)
+			},
+		},
+		"cpu explicitly overridden by zero value": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(1024)
+				svc.Environments["test"].CPU = aws.Int(0)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(0)
+			},
+		},
+		"cpu not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(1024)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(1024)
+			},
+		},
+		"memory overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Memory = aws.Int(1024)
+				svc.Environments["test"].Memory = aws.Int(2048)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.Memory = aws.Int(2048)
+			},
+		},
+		"memory explicitly overridden by zero value": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Memory = aws.Int(1024)
+				svc.Environments["test"].Memory = aws.Int(0)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.Memory = aws.Int(0)
+			},
+		},
+		"memory not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(1024)
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.CPU = aws.Int(1024)
+			},
+		},
+		"platform overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Platform = &PlatformArgsOrString{
+					PlatformString: aws.String("mockPlatform"),
+				}
+				svc.Environments["test"].Platform = &PlatformArgsOrString{
+					PlatformString: aws.String("mockPlatformTest"),
+				}
+			},
+			wanted: func(svc *LoadBalancedWebService) {
+				svc.Platform = &PlatformArgsOrString{
+					PlatformString: aws.String("mockPlatformTest"),
+				}
+			},
+		},
+		"platform not overridden": {
+			inSvc: func(svc *LoadBalancedWebService) {
+				svc.Platform = &PlatformArgsOrString{
```

Comment:
Do we not need to test if `PlatformString` can override the args like https://github.com/aws/copilot-cli/pull/2728/files#diff-b30de9ebe3c4859ebd00dbc4695a04c836597aff05a1fdc25cb2cc923ff83037R279

Reply:
We do! It's going to be added here: https://github.com/aws/copilot-cli/blob/ed91dbc861196fe5c156cba07043d79291f78d33/internal/pkg/manifest/applyenv_test.go#L1106